src/ray/object_manager/connection_pool.cc

```diff
@@ -0,0 +1,165 @@
+#include "connection_pool.h"
+
+namespace ray {
+
+ConnectionPool::ConnectionPool(ObjectDirectoryInterface *object_directory,
+                               asio::io_service *connection_service,
+                               const ClientID &client_id) {
+  object_directory_ = object_directory;
```

Comment:
If the goal is to eventually make the `ConnectionPool` thread-safe, I don't think we should keep a pointer to the `ObjectDirectoryInterface` in this class. Let's try to look up the address that we need to connect to outside of the `ConnectionPool`.

Reply:
See comment in #1791 regarding `ObjectDirectory`.